File: solutions/go/proverb/1/proverb.go

// Package proverb generates lines of a poem for a string array
package proverb

import ( "fmt")

// The Proverb function will return lines of poems giving the rhyme string array
func Proverb(rhyme []string) []string {
	var poem []string
	if len(rhyme) == 0 {
		return []string{}
	}
	word1 := rhyme[0]
	
	for wi := 1; wi < len(rhyme); wi++ {
		word2 := rhyme[wi]
		poem = append(poem, fmt.Sprintf("For want of a %v the %v was lost.", word1, word2))
		word1 = word2
	}
	poem = append(poem, fmt.Sprintf("And all for the want of a %v.", rhyme[0]))
	return poem
}

File: solutions/csharp/proverb/1/Proverb.cs
using System;
using System.Collections.Generic;

public static class Proverb
{
    public static string[] Recite(string[] subjects)
    {
        List<string> pb = new List<string>();

        for (int si = 0; si<subjects.Length-1; si++)  {

            pb.Add($"For want of a {subjects[si]} the {subjects[si+1]} was lost.");

        }
        if (subjects.Length>0) pb.Add($"And all for the want of a {subjects[0]}.");
        return pb.ToArray();
    }
}